src/controllers/controller.bar.js

```diff
@@ -164,7 +164,7 @@ function computeFlexCategoryTraits(index, ruler, options) {
 	if (prev === null) {
 		// first data: its size is double based on the next point or,
 		// if it's also the last data, we use the scale end extremity.
-		prev = curr - (next === null ? ruler.end - curr : next - curr);
+		prev = curr - (next === null ? Math.max(curr - ruler.start, ruler.end - curr) * 2 : next - curr);
```

Comment:
This is getting a bit tricky to follow. I feel like we're trying to hack `prev` and `next` to get the desired results for `start` and `size`. I wonder if it would be easier if we just computed those directly like:

```
if (prev == null && next == null) {
  start = ruler.start * (1 - percent / 2);
  size = ruler.end - ruler.start;
} else if (prev == null) {
  start = curr - ((next - curr) / 2) * percent;
  size = ((next - curr) / 2) * percent;
} else if (next == null) {
  start = curr - ((curr - prev) / 2) * percent;
  size =  (curr - prev) * percent;
} else {
  start = curr - ((curr - prev) / 2) * percent;
  size = ((next - prev) / 2) * percent;
}
```

Note that these formulas probably aren't at all correct. This is just to demonstrate an alternate way of laying out the code

Reply:
@benmccann your proposal introduces lot of code duplication and actually doesn't look clearer to me since it's hard to follow all the `if` / `else`. The current implementation is not a "hack" of prev / next but an adjustment when they are not defined. Comments explain pretty well what's going on so please don't reformat this part of the code.